azure/services/loadbalancers/spec.go

```diff
@@ -263,7 +263,7 @@ func getProbes(lbSpec LBSpec) []*armnetwork.Probe {
 					Port:              ptr.To[int32](lbSpec.APIServerPort),
 					RequestPath:       ptr.To(httpsProbeRequestPath),
 					IntervalInSeconds: ptr.To[int32](15),
-					NumberOfProbes:    ptr.To[int32](4),
+					ProbeThreshold:    ptr.To[int32](4),
```

Comment:
```suggestion
					ProbeThreshold:    ptr.To[int32](1),
```

`NumberOfProbes` is ignored by Azure, and the default value for `ProbeThreshold` is `1`, so CAPZ has effectively been using a `ProbeThreshold` == 1 for some time. Increasing the ProbeThreshold would mean that the load balancer continues to send requests to an API server longer after it has shut down. So let's set this to 1.

Reply:
Fixed.